packages/sheets-ui/src/facade/f-range.ts

```diff
@@ -179,11 +179,19 @@ class FRangeSheetsUIMixin extends FRange implements IFRangeSheetsUIMixin {
     }
 
     override generateHTML(): string {
-        const copyContent = this._injector.get(ISheetClipboardService).generateCopyContent(
+        const clipboardService = this._injector.get(ISheetClipboardService);
+        const hooks = clipboardService.getClipboardHooks();
+        const unitId = this._workbook.getUnitId();
+        const subUnitId = this._worksheet.getSheetId();
+        const range = this._range;
+
+        hooks.forEach((h) => h.onBeforeCopy?.(unitId, subUnitId, range, COPY_TYPE.COPY));
```

Comment:
We should not expose implementation details in the Facade layer, maybe there should be a `generateCopyContent` method in the clipboard service to invoke the hooks and do other things.